### PR TITLE
Upgrade to Akka 2.4.6 #15

### LINKED ISSUE #15: please, update to latest version of akka stream
Hi, 
is it possible to update this example with the latest version of akka.
A lot of method (as onComplete) are not yet supported.

Also BasicTransformation not compile...

Thanks and regards

--
matpil

### PR DESCRIPTION
**Pull request by johanandren**
Fixes #15

### CODE REVIEW

File: src/main/java/sample/stream/GroupLogFile.java [hunk 1]
```diff
@@ -32,37 +30,35 @@ public static void main(String[] args) throws IOException {
...
+      // write lines of each group to a separate file
+      runForeach((levelProducerPair) -> {
+        outputs.get(levelProducerPair.first()).println(levelProducerPair.second());
+      }, materializer).handle((ignored, failure) -> {
+        outputs.forEach((key, writer) -> {
+          writer.close();
+        });
+        system.terminate();
+         return NotUsed.getInstance();
+      });
```
Comment (ktoso) [inline src/main/java/sample/stream/GroupLogFile.java]: oh, sneaky - avoiding the groupby hm hm.
I'd add a ticket though that the actual nice way we want is still impossible (needs that other version of groupBy)

Comment (johanandren) [inline src/main/java/sample/stream/GroupLogFile.java]: Yeah, I thought that it was more important to get this up to 2.4.6 than to wait for a new way to do the exact same, in this use case it wasn't very useful anyways.

Comment (ktoso) [inline src/main/java/sample/stream/GroupLogFile.java]: Yeap, and I think that's good 👍 
Do we have a ticket for the "good" way?

Comment (johanandren) [inline src/main/java/sample/stream/GroupLogFile.java]: Yup https://github.com/akka/akka/issues/18969

Comment (ktoso) [inline src/main/java/sample/stream/GroupLogFile.java]: Good, LGTM :)

**Comment by ktoso** [comment]
LGTM

**Comment by drewhk** [comment]
LGTM!

**Pull request merged**